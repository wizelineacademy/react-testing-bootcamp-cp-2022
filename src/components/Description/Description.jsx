import styled from "styled-components";

const Description = ({ explanation }) => (
	<DescriptiveParagraph data-testid="description">
		{explanation}
	</DescriptiveParagraph>
);

const DescriptiveParagraph = styled.p`
	margin-top: 85px;
	text-justify: distribute;
	line-height: 30px;
`;

export default Description;
