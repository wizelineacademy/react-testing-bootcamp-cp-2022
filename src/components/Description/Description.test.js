import { render, screen } from "@testing-library/react";

import Description from ".";

const setUp = (props) => render(<Description {...props} />);

describe("Description", () => {
	describe("when a description prop is passed", () => {
		it("should display said description", () => {
			const props = {
				explanation:
					"The calendar month, a full moon name, distance in kilometers, and angular size is indicated for each stripe.",
			};
			const { container } = setUp(props);

			const description = screen.getByText(props.explanation);

			expect(description).toBeInTheDocument();
			expect(container).toMatchSnapshot();
		});
	});
});
