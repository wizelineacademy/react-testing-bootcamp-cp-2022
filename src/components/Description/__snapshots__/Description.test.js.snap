// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description when a description prop is passed should display said description 1`] = `
<div>
  <p
    class="sc-bczRLJ cPltQN"
    data-testid="description"
  >
    The calendar month, a full moon name, distance in kilometers, and angular size is indicated for each stripe.
  </p>
</div>
`;
